app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListView.kt

```diff
@@ -24,37 +24,55 @@ class PromotedStoryListView @JvmOverloads constructor(
 
   @Inject
   lateinit var bindingInterface: ViewBindingShim
+  private var dataList: List<PromotedStoryViewModel> = listOf()
 
   override fun onAttachedToWindow() {
     super.onAttachedToWindow()
 
     (FragmentManager.findFragment<Fragment>(this) as ViewComponentFactory)
       .createViewComponent(this).inject(this)
 
-    adapter = BindableAdapter.SingleTypeBuilder.newBuilder<PromotedStoryViewModel>()
+    /**
+     * The StartSnapHelper is used to snap between items rather than smooth scrolling, so that
+     * the item is completely visible in [HomeFragment] as soon as learner lifts the finger
+     * after scrolling.
+     */
+    val snapHelper = StartSnapHelper()
+    this.onFlingListener = null
+    snapHelper.attachToRecyclerView(this)
+  }
+
+  /**
+   * Sets the list of promoted stories that this view shows to the learner.
+   *
+   * @param newDataList the new list of stories to present
+   */
+  fun setDataList(newDataList: List<PromotedStoryViewModel>) {
+    // Update the adapter and the story list only if the list is new. The parent presenter should
+    // not render promoted stories if the list is empty, but default to showing the last list.
+    if (newDataList != null && !newDataList.isEmpty() && newDataList != dataList) {
```

Comment:
Should we be checking emptiness in newDataList? Isn't it actually valid to sometimes bind empty (e.g. if there are no promoted stories)?

Reply:
We shouldn't bind an empty list as we just don't render this view if there is no list -- will do error checking here instead.

Reply:
Done

Added a warning here but didn't make it blocking, let me know if this isn't reasonable

Reply:
I'm not convinced about the 'we shouldn't bind an empty list' part, but I left a newer comment to discuss this.

The warning otherwise seems fine to me.